src/azure/cli/_output.py

```diff
@@ -125,6 +115,42 @@ def out(self, obj):
     def get_formatter(format_type):
         return OutputProducer.format_dict.get(format_type, format_list)
 
+class TableOutput(object): #pylint: disable=too-few-public-methods
+
+    SKIP_KEYS = ['id', 'type']
+
+    @staticmethod
+    def _capitalize_first_char(x):
+        return x[0].upper() + x[1:] if x and len(x) > 0 else x
+
+    @staticmethod
+    def _auto_table_item(item):
+        new_entry = OrderedDict()
+        for k in item.keys():
+            if k in TableOutput.SKIP_KEYS:
+                continue
+            if item[k] and not isinstance(item[k], (list, dict, set)):
+                new_entry[TableOutput._capitalize_first_char(k)] = item[k]
+        return new_entry
+
+    @staticmethod
+    def _auto_table(result):
+        if isinstance(result, list):
+            new_result = []
+            for item in result:
+                new_result.append(TableOutput._auto_table_item(item))
+            return new_result
+        else:
+            return TableOutput._auto_table_item(result)
+
+    @staticmethod
+    def dump(data):
+        table_data = TableOutput._auto_table(data)
+        table_str = tabulate(table_data, headers="keys", tablefmt="simple") if table_data else ''
+        if table_str == '\n':
+            raise ValueError('Unable to extract fields for table.')
```

Comment:
Is this what we talked about at the meeting, where if there is data but the extracted table is empty we throw an error?